applications/tari_watcher/src/config.rs

```diff
@@ -0,0 +1,180 @@
+
+    /// The path of the validator node registration file, containing signed information required to
+    /// submit a registration transaction on behalf of the node
+    pub vn_registration_file: PathBuf,
+
+    /// The sidechain ID to use. If not provided, the default Tari sidechain ID will be used.
+    pub sidechain_id: Option<String>,
+
+    /// The configuration for managing one or multiple processes
+    pub instance_config: Vec<InstanceConfig>,
+
+    /// The process specific configuration for the executables
+    pub executable_config: Vec<ExecutableConfig>,
+
+    /// The channel configuration for alerting and monitoring
+    pub channel_config: Vec<ChannelConfig>,
+}
+
+impl Config {
+    pub(crate) async fn write<W: io::AsyncWrite + Unpin>(&self, mut writer: W) -> anyhow::Result<()> {
+        let toml = toml::to_string_pretty(self)?;
+        writer.write_all(toml.as_bytes()).await?;
+        Ok(())
+    }
+}
+
+#[derive(Debug, Clone, Copy, Eq, PartialEq, serde::Serialize, serde::Deserialize)]
+pub enum InstanceType {
+    TariValidatorNode,
+    MinoTariConsoleWallet,
+}
+
+impl Display for InstanceType {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        fmt::Debug::fmt(self, f)
+    }
+}
+
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+pub struct ChannelConfig {
+    pub name: String,
+    pub enabled: bool,
+    pub credentials: String,
+}
+
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+pub struct ExecutableConfig {
+    pub instance_type: InstanceType,
+    pub executable_path: Option<PathBuf>,
+    pub compile: Option<CompileConfig>,
+    pub env: Vec<(String, String)>,
+}
+
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+pub struct CompileConfig {
+    pub working_dir: Option<PathBuf>,
+    pub package_name: String,
+    pub target_dir: Option<PathBuf>,
+}
+
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+pub struct InstanceConfig {
+    pub name: String,
+    pub instance_type: InstanceType,
+    pub num_instances: u32,
+    #[serde(alias = "extra_args")]
+    pub settings: HashMap<String, String>,
+}
+
+impl InstanceConfig {
+    pub fn new(instance_type: InstanceType) -> Self {
+        Self {
+            name: instance_type.to_string(),
+            instance_type,
+            num_instances: 1,
+            settings: HashMap::new(),
+        }
+    }
+
+    pub fn with_name<S: Into<String>>(mut self, name: S) -> Self {
+        self.name = name.into();
+        self
+    }
+
+    pub fn with_num_instances(mut self, num_instances: u32) -> Self {
+        self.num_instances = num_instances;
+        self
+    }
+}
+
+pub fn get_base_config(cli: &Cli) -> anyhow::Result<Config> {
+    let executables = vec![
+        ExecutableConfig {
+            instance_type: InstanceType::TariValidatorNode,
+            executable_path: Some("target/release/minotari_node".into()),
+            compile: Some(CompileConfig {
```

Comment:
We don't need to compile binaries in the watcher. We can assume `tari_validator_node` and `minotari_console_wallet` are in the PATH. We should allow the user to override the VN binary path.